sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -111,6 +113,35 @@ private[hive] object HiveShim {
     }
   }
 
+  private def isSubClassOf(t: Type, parent: Class[_]): Boolean = t match {
+    case cls: Class[_] => parent.isAssignableFrom(cls)
```

Comment:
Will you pass in something not a `Class[_]`? If not, we can simply inline `isAssignableFrom` check into `hasInheritanceOf`.